File: repos/BusinessService/Models/Entities/Course.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using BusinessService.Models.DTOs;

namespace BusinessService.Models.Entities
{
    [Table("Courses")]
    public class Course
    {
        [Key]
        [Column("courseId")]
        public Guid CourseId { get; set; }

        [Column("tutorFk")]
        public Guid TutorId { get; set; }
        [Column("price")]
        public decimal Price { get; set; }
        [Column("title")]
        public string? Title { get; set; }
        [Column("introduction")]
        public string? Introduction { get; set; }
        [Column("description")]
        public string? Description { get; set; }
        [Column("courseDifficultyFk")]
        public Guid? CourseDifficultyId { get; set; }
        [Column("categoryFk")]
        public Guid? CategoryId { get; set; }
        [Column("createdDate")]
        public DateTime? CreatedDate { get; set; }
        [Column("updatedDate")]
        public DateTime? UpdatedDate { get; set; }
        [Column("publishedDate")]
        public DateTime? PublishedDate { get; set; }
        [Column("isEnabled")]
        public bool? IsEnabled { get; set; } = false;
        [Column("courseImage")]
        public byte[]? CourseImage {  get; set; }
        [Column("languageFk")]
        public Guid? LanguageFk { get; set; }
        [Column("tags")]
        public List<string>? Tags { get; set; }
        [Column("isDeleted")]
        public bool? IsDeleted { get; set; } = false;

        [Column("currency")]
        public string? Currency { get; set; }
        public Category? Category { get; set; }
        public CourseDifficulty? CourseDifficulty { get; set; }
        public ICollection<CourseContent> CourseContent { get; set; } = new List<CourseContent>();
        public List<Enrollment> Enrollment { get; set; }
    }
}

File: frontend/src/types/course.ts
// export interface Course {
//        courseid: number ;
//        courseName: string ;
//        courseIntro:string;
//        price: number;
//        categoryName : string;
//        PublishedDate: Date;
//        courseContentId: string;
//        courseDescription: string;
//        briefIntro: string;
//        resources: string | null;
//        courseLink: string;
//        level : string;
//        duration: string;
// }

export interface PaginatedCourse {
  courses: TutorCourse[];
  page: number;
  pageSize: number;
  totalItems: number;
}

export interface PaginatedCourses {
  courses: TutorCourse[];
  page: number;
  pageSize: number;
  totalItems: number;
}

export interface Course {
  courseId: string;
  title: string;
  description: string;
  introduction: string;
  courseDifficultyName: 'Beginner' | 'Intermediate' | 'Advanced';
  categoryName: string;
  tutorId: string;
  tutorName: string;
  courseImage: string;
  price: number;
  isEnabled: boolean;
  enrolledStudents: number;
  rating: number;
  reviewCount: number;
  createdDate: Date;
  updatedDate: Date;
  totalLessons: number;
  totalDuration: string; // e.g., "10h 30m"
  languages: string[];
  tags: string[];
  contents?: CourseContent[];
  currency: string;
}

// Course content types for tutor course management
export interface SubContent {
  subContentId: string;
  subContentOrder: number;
  subContentTitle: string;
  subContentDescription: string;
  type: 'video' | 'document' | 'both';
  videoFile?: File | null;
  documentFile?: File | null;
  videoUrl?: string; // For display purposes
  documentUrl?: string; // For display purposes
}

export interface CourseContent {
  contentId: string;
  contentTitle: string;
  contentDuration: string; // e.g., "45 min", "1.5 hours"
  contentDescription: string;
  contentSortOrder: number;
  subContent: SubContent[];
}

// Enhanced course interface for tutor course management
export interface TutorCourse {
  courseId: string;
  title: string;
  description: string;
  introduction: string;
  courseDifficultyName: 'Beginner' | 'Intermediate' | 'Advanced';
  courseDifficultyFk: '';
  categoryFk: '';
  categoryName: string;
  tutorId: string;
  tutorName: string;
  courseImage: string;
  price: number;
  isEnabled: boolean;
  enrolledStudents: number;
  rating: number;
  reviewCount: number;
  createdDate: Date;
  updatedDate: Date;
  totalLessons: number;
  totalDuration: string; // e.g., "10h 30m"
  languages: string;
  languageFk: '';
  tags: string[];
  currency: string;
  courseContent: CourseContent[]; // Optional course content
}
